### ISSUE TITLE
[Feature] Nuget source "github"

### ISSUE THREAD

**Issue by MoazAlkharfan** [rohith/publish-nuget#21]
With the new github package registry i think this would be a good feature to have.

**Comment by rohith** [comment]
Yeah I tried it before but the `dotnet` cli didn't support GPR. I need to see if something changed with the GPR support

If you're aware of this & think that it's resolved then please feel free to create a PR

**Comment by alexrosenfeld10** [comment]
Any plans to make the source not just nuget or github, but parameterizable? I'm looking to use this to publish to a private nuget repository, like my own artifactory.

**Comment by AraHaan** [comment]
me I would like a config varaible like so:

```yml
NUGET_SOURCE: <url source>
NUGET_SYMBOL_SOURCE: <url symbol source>
```

``NUGET_SYMBOL_SOURCE`` would then be used for symbol packages (``symbols.nupkg`` / ``*.snupkg``).

**Comment by rohith** [comment]
@alexrosenfeld10 @AraHaan yup I'll push an update sometime soon

**Comment by rohith** [comment]
@MoazAlkharfan I'm not quite sure on the support of GPR yet, I believe the issue while pushing a package is resolved recently but this action uses `https://api.nuget.org/v3-flatcontainer/` for version change detection & I'm yet to find a way to do that with GPR. Any suggestions are welcome

I'll anyways update the action to parameterize the NuGet feed url & it should work with feeds that are either similar to or built with NuGet

**Comment by rohith** [comment]
@alexrosenfeld10 @AraHaan `v2.4.0` has custom NuGet feed support, it only works with feeds that have `v3-flatcontainer/` endpoint due to the way this action does version change detection

**Comment by alexrosenfeld10** [comment]
@rohith thanks I actually ended up rolling my own, it does a bunch of other behavior I was also looking for.

**Comment by Rutgerz** [comment]
I was also hoping to use this action to publish to github packages and ended up having a look around the  [Nuget API spec ](https://docs.microsoft.com/en-us/nuget/api/overview). 
Turns out Github packages does support the required endpoint, but the URL is different. The `v3-flatcontainer/` that is being queried is called the [PackageBaseAddress](https://docs.microsoft.com/en-us/nuget/api/package-base-address-resource). Having this endpoint is required for any nuget server, however the URL for it is specified in the [Service Index](https://docs.microsoft.com/en-us/nuget/api/overview#service-index). 
So what would have to changes is that first the `NugetSource/index.json` is queried and from that response the PackageBaseAddress Resource Url found and then that can be queried as before. 


## Nuget.org
```http
GET https://api.nuget.org/v3/index.json
```
 returns
```jsonc
{
  "version": "3.0.0",
  "resources": [
    // removed for brevity...
    {
      "@id": "https://api.nuget.org/v3/registration5-semver1/",
      "@type": "RegistrationsBaseUrl",
      "comment": "Base URL of Azure storage where NuGet package registration info is stored"
    },
    {
      "@id": "https://api.nuget.org/v3-flatcontainer/",
      "@type": "PackageBaseAddress/3.0.0",
      "comment": "Base URL of where NuGet packages are stored, in the format https://api.nuget.org/v3-flatcontainer/{id-lower}/{version-lower}/{id-lower}.{version-lower}.nupkg"
    },
    // more removed for brevity
  ],
  "@context": {
    "@vocab": "http://schema.nuget.org/services#",
    "comment": "http://www.w3.org/2000/01/rdf-schema#comment"
  }
}
```

This tells us to use `https://api.nuget.org/v3-flatcontainer/`. Github will return similar response:

## Github Packages
```http
GET https://nuget.pkg.github.com/OWNER/index.json
Authorization: Bearer TOKEN
```
returns
```jsonc
{
      "version": "3.0.0-beta.1",
      "resources": [
        {
          "@id": "https://nuget.pkg.github.com/OWNER/download",
          "@type": "PackageBaseAddress/3.0.0",
          "comment": "Get package content (.nupkg)."
        },
        // removed
      ]
    }
```
So `https://nuget.pkg.github.com/OWNER/download` functions the same as the v3-flatcontainer on nuget.org.

Hopefully this makes sense. I felt like it would be a waste to not at least try to dump this information here after looking into it. I've never used node.js before myself, otherwise I would have attempted a quick PR (and maybe I still might)

**Comment by brandedoutcast** [comment]
@Rutgerz Thank you that's a great news & such a detailed comment. I'll wait for that PR & if not I'll update the action to support GPR

**Comment by brgrz** [comment]
It'd be awesome if this action could also publish to GitHub Packages. @jcansdale from GitHub is the GPR maintainer, maybe he can help?

**Comment by jcansdale** [comment]
Some notes when _publishing_ to GitHub Packages:

1. The project's `RepositoryUrl` must point to a GitHub repository URL
2. The `OWNER` in `https://nuget.pkg.github.com/OWNER/index.json` can be anything
3. Basic authentication must be used with a PAT as the password (the username can be anything)

Yes, it's a little strange. 😉

**Comment by fasetto** [comment]
https://github.com/brandedoutcast/publish-nuget/blob/c12b8546b67672ee38ac87bea491ac94a587f7cc/index.js#L90-L115
I think there is no need to use *flatcontainer* endpoint. Push operating will fail anyways if the package with provided version already exists on the nuget.

**Comment by Rebel028** [comment]
So since there's no progress on this issue you can try using my fork of this action: https://github.com/marketplace/actions/publish-nuget-2 

I made it work in my case (private team repo) so maybe it will help some of you.

It also has the feature described in #43

**Comment by Rutgerz** [comment]
Just as an update, I not longer have real need for this issue. I can now succesfully push (and restore from) github packages by using https://github.com/actions/setup-dotnet. Here an example of what I run now
```yaml
```
